src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,52 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            ReadOnlySpan<byte> inputValue = value;
+
+            // We will ignore the leading and trailing occurrences of the white space in the json check.
+            int start = 0;
+            while (start < inputValue.Length && char.IsWhiteSpace(Convert.ToChar(inputValue[start])))
+            {
+                start++;
+            }
+            int end = inputValue.Length - 1;
+            while (end >= start && char.IsWhiteSpace(Convert.ToChar(inputValue[end])))
+            {
+                end--;
+            }
+            int len = end - start + 1;
+
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json string, Json value true, Json value false, Json number.
+            // Our logic here is that, for example, we check if the first and the last bytes are '{' and '}', respectively.
+            if (len > 0 && (
+                inputValue[start] == (byte)'{' && inputValue[end] == (byte)'}' ||
+                inputValue[start] == (byte)'[' && inputValue[end] == (byte)']' ||
+                inputValue[start] == (byte)'"' && inputValue[end] == (byte)'"' ||
+                inputValue[start] == (byte)'t' && inputValue[end] == (byte)'e' ||
```

Comment:
What if my value is:

- telephone
- fortnite
- 9one1
- ["invalid]
- {"no]}

Unless I'm misreading, these would all pass this validation.

Reply:
The first three bullets are the same comment [here](https://github.com/Azure/autorest.csharp/pull/2544#discussion_r943715228).  The last two would be handled by the catch block and written as `"["invalid]"` and `"{"no]}"` respectively.  Again the pre-parse is not meant to be 100% accurate but its simply meant to lower our average case by eliminating things that we know for sure are NOT valid json.